File: src/pucrs/domain/PosicaoDeMemoria.cs
using System;

namespace Modelos
{
    public class PosicaoDeMemoria
    {
        public string OPCode { get; set; }

        public string Reg1 { get; set; }

        public string Reg2 { get; set; }

        public int Parameter { get; set; }

        public override string ToString()
        {
            if (OPCode == "DATA")
            {
                return $"OPCode: {OPCode}\nParameter: {Parameter}";
            }

            return $"OPCode: {OPCode}\nRegistrador 1: {Reg1}\nRegistrador 2: {Reg2}\nParameter: {Parameter}";
        }
    }
}

File: src/pucrs/domain/PosicaoDeMemoria.java
package pucrs.domain;

public class PosicaoDeMemoria {
    public String opCode;
    public String reg1;
    public String reg2;
    public int parameter;

    //FAZER @OVERRIDE TOSTRING

    public String getOpcode() {
        return this.opCode;
    }

    public void setOpcode(String opCode) {
        this.opCode = opCode;
    }

    public String getReg1() {
        return this.reg1;
    }

    public void setReg1(String reg1) {
        this.reg1 = reg1;
    } 

    public String getReg2() {
        return this.reg2;
    }

    public void setReg2(String reg2) {
        this.reg2 = reg2;
    } 

    public int getParameter() {
        return this.parameter;
    }

    public void setParameter(int parameter) {
        this.parameter = parameter;
    } 
}